### ISSUE TITLE
MaxPooling2D fails to utilize last column in odd dimensional input

### ISSUE THREAD

**Issue by mathDR** [fchollet/keras#2094]
When calling MaxPooling2D with pool_size=(2,2) on inputs with an odd dimension (say 1x4x5) the output dimension is 1x2x2.  Thus the pooling operator is not utilizing the information in the last column of the input.  The code that produces the size of the output:  conv_output_length does an integer divide operation giving the floor of the required number of dimensions.

How can I rectify this?  The simplest idea I have would be to zero pad another column making the input (1x4x6) and not messing with the pooling operation at all, but I don't think non-symmetric padding is allowed in keras.  Is this right?

**Comment by mathDR** [comment]
Seems like almost a duplicate of (closed) #583.  Advice there is to roll my own change of the ZeroPadding2D layer
